### PR TITLE
Rest API for retrieve allusers

### PR DESCRIPTION
**Pull request by EXR-86**

### CODE REVIEW

File: src/main/java/com/elixr/poc/service/UserRetrievalService.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        this.userRepository = userRepository;
+        this.conversionService = conversionService;
+    }
+
+    /**
+     * Retrieving all users
+     * @return
+     */
+    public List<User> getAllUsers(){
+        List<User>  allUsers = userRepository.findAll();
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/service/UserRetrievalService.java]: "return userRepository.findAll()" single statement is enough

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import com.elixr.poc.service.UserRetrievalService;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/home")
+public class UserRetrievalController {
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java:14]: Add proper class level comments

File: src/main/java/com/elixr/poc/rest/response/UserGetResponse.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+import com.elixr.poc.data.User;
+import jakarta.validation.constraints.Negative;
+import lombok.Builder;
+import lombok.Data;
+
+import java.util.List;
+
+@Data
+@Builder
+public class UserGetResponse {
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/rest/response/UserGetResponse.java]: UserGetResponse & UserPostResponse has a common field boolean success
So its better to define an abstract class UserReponse with 1 attribute "protected boolean success"
Then UserGetResponse & UserPostResponse extends this new class

**Comment by kssreejith1994** [comment]
@ashutosh2406 please resolve the conflicts.

File: src/main/java/com/elixr/poc/service/UserRetrievalService.java (hunk 1 above)
Comment (EXR-86) [reply src/main/java/com/elixr/poc/service/UserRetrievalService.java]: changed to  return userRepository.findAll()

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java (hunk 2 above)
Comment (EXR-86) [reply src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java:14]: Added propper class level comments

File: src/main/java/com/elixr/poc/rest/response/UserGetResponse.java (hunk 3 above)
Comment (EXR-86) [reply src/main/java/com/elixr/poc/rest/response/UserGetResponse.java]: Created a class UserResponse and extending it where needed

File: src/main/java/com/elixr/poc/rest/response/UserResponse.java [hunk 4]
```diff
@@ -0,0 +1,11 @@
+package com.elixr.poc.rest.response;
+
+
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+
+@AllArgsConstructor
+@NoArgsConstructor
+public abstract class UserResponse {
+    public boolean success;
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/rest/response/UserResponse.java]: make this protected

File: src/main/java/com/elixr/poc/service/UserService.java [hunk 5]
```diff
@@ -49,4 +52,14 @@ private User saveDataToDatabase(User user) {
...
         return user;
     }
+
+
+    /**
+     * Retriving all the users
+     * @return
+     */
+    public UserGetResponse getAllUsers() {
+
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/service/UserService.java]: remove empty line

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+
+    public UserRetrievalController(UserService userService) {
+        this.userService = userService;
+    }
+
+    @GetMapping("/users")
+    public UserGetResponse getAllUser() {
+        return userService.getAllUsers();
+    }
+}
```
Comment (mh-sraveendran) [inline src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java:26]: checkstyle error.
add a empty line at EOF.

File: src/main/java/com/elixr/poc/service/UserService.java (hunk 5 above)
Comment (EXR-86) [reply src/main/java/com/elixr/poc/service/UserService.java]: removed empty line

File: src/main/java/com/elixr/poc/rest/response/UserResponse.java (hunk 4 above)
Comment (EXR-86) [reply src/main/java/com/elixr/poc/rest/response/UserResponse.java]: made that as protected

File: src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java (hunk 6 above)
Comment (EXR-86) [reply src/main/java/com/elixr/poc/rest/controller/UserRetrievalController.java:26]: added empty line at EOF

**Pull request merged**